tests/operators/test_regrid.py

```diff
@@ -152,10 +153,53 @@ def test_regrid_onto_xyspacing_unknown_method(regrid_source_cube):
 
 
 @pytest.mark.filterwarnings("ignore:Selected point is within")
-def test_regrid_to_single_point(cube):
+def test_regrid_to_single_point_east(cube):
     """Regrid to single point."""
     # Test extracting a single point.
-    regrid_cube = regrid.regrid_to_single_point(cube, 0.5, 358.5, "Nearest")
+    # Test that the grid latitude rotation works when the
+    # centre of the grid is too far east (note that the
+    # test cube, by default, is too far east, centred east
+    # of 180 deg).
+    cube_fix = read._longitude_fix_callback(cube, None, None)
+    regrid_cube = regrid.regrid_to_single_point(
+        cube_fix, 0.5, -1.5, "Nearest", boundary_margin=1
+    )
+    expected_cube = "<iris 'Cube' of air_temperature / (K) (time: 3)>"
+    assert repr(regrid_cube) == expected_cube
+
+
+def test_regrid_to_single_point_west(cube):
+    """Regrid to single point."""
+    # Test extracting a single point.
+    # Test that the grid latitude rotation works when the
+    # centre of the grid is too far west.
+    long_coord = cube.coord("grid_longitude").points.copy()
+    long_coord -= 720.0
+    cube.coord("grid_longitude").points = long_coord
+    cube_fix = read._longitude_fix_callback(cube, None, None)
+    regrid_cube = regrid.regrid_to_single_point(
+        cube_fix, 0.5, -1.5, "Nearest", boundary_margin=1
+    )
+    expected_cube = "<iris 'Cube' of air_temperature / (K) (time: 3)>"
+    assert repr(regrid_cube) == expected_cube
+
+
+def test_regrid_to_single_point_longitude_transform_1(cube):
+    """Regrid to single point."""
+    # Test extracting a single point.
+    regrid_cube = regrid.regrid_to_single_point(
+        cube, 0.5, 358.5, "Nearest", boundary_margin=1
+    )
+    expected_cube = "<iris 'Cube' of air_temperature / (K) (time: 3)>"
+    assert repr(regrid_cube) == expected_cube
+
+
+def test_regrid_to_single_point_longitude_transform_2(cube):
+    """Regrid to single point."""
+    # Test extracting a single point.
```

Comment:
Description into docstring. Could also add a bit about what differentiates this test and the previous one.

Reply:
The text has been moved into the doc-string and description made to read more clearly.